packages/grafana-data/src/types/query.ts

```diff
@@ -52,3 +52,62 @@ export interface DataQuery {
    */
   datasource?: DataSourceRef | null;
 }
+
+/**
+ * Abstract representation of any label-based query
+ * @internal
+ */
+export interface AbstractQuery extends DataQuery {
+  labelMatchers: AbstractLabelMatcher[];
+}
+
+/**
+ * @internal
+ */
+export enum AbstractLabelOperator {
```

Comment:
should we consider adding any additional operators here even if core data sources don't currently support them? enterprise data sources won't be able to easily upgrade dependencies, so there's a possibility they could be pinned to the types available in the initial release of this feature for an extended time.

Reply:
Adding more operators should be backward compatible, so I think we could just backport new operators to current stable. Patch releases are every week so plugins wouldn't need to wait for more than a week to upgrade. BTW. What's the current process of updating dependencies in enterprise plugins?

Reply:
> I think we could just backport new operators to current stable

if it was backported from 8.5.0 to 8.4.1, and the customer was on 8.4.0, i think we'd hit a runtime error due to the API change.

> What's the current process

this is a hard/complex question to answer, but in general right now enterprise plugins need to maintain support for the previous major version of grafana, so a good portion of enterprise plugins are pinned to 7.x dependencies.

> Patch releases are every week so plugins wouldn't need to wait for more than a week

would additions need to wait for the next minor?

Reply:
Guess it's a separate discussion @toddtreece but wouldn't we just say sorry the feature isn't available on old Grafana so it will be only in new version of the enterprise data source and customer will have to upgrade?

Reply:
I think the point is that we want to be able to use this feature in Enterprise plugins, but with the requirement to support one major version back it's a bit harder. On the other hand the only common API here is AbstractQuery object (I don't expect method signatures to change). In worst case we case make the object a bit more flexible allowing custom key/value strings to avoid any TS errors in case API is extended in the future. I'll merge it the way it is at the moment, but we can revise it before 8.4.0